torch/fft/__init__.py

```diff
@@ -87,6 +87,98 @@
+    representation where only the positive frequencies of the last dimension
+    are stored explicitly.
+
+Args:
+    input (Tensor): the input tensor
+    s (Tuple[int], optional): Signal size in the transformed dimensions. If
+        given, the input will either be zero-padded or trimmed to this length
+        before computing the FFT.
+    dim (Tuple[int], optional): Dimensions to be Fourier transformed.
+        Default: all dimensions.
+    norm (str, optional): Normalization mode. For the forward transform
+        (:func:`~torch.fft.fftn`), these correspond to:
+
+        * ``"forward"`` - normalize by ``1/n``
+        * ``"backward"`` - no normalization
+        * ``"ortho"`` - normalize by ``1/sqrt(n)`` (making the FFT orthonormal)
+
+        Where ``n = prod(s)`` is the logical FFT size.
+        Calling the backward transform (:func:`~torch.fft.ifftn`) with the same
+        normalization mode will apply an overall normalization of ``1/n``
+        between the two transforms. This is required to make
+        :func:`~torch.fft.ifftn` the exact inverse.
+
+        Default is ``"backward"`` (no normalization).
+
+Example:
+
+    >>> import torch.fft
+    >>> x = torch.rand(10, 10, dtype=torch.complex64)
+    >>> fftn = torch.fft.fftn(t)
+
+    The Fourier transform is separable, so :func:`~torch.fft.fftn` here is
+    equivalent to two one-dimensional :func:`~torch.fft.fft` calls:
+
+    >>> two_ffts = torch.fft.fft(torch.fft.fft(x, dim=0), dim=1)
+    >>> torch.allclose(fftn, two_ffts)
+
+""")
+
+ifftn = _add_docstr(_fft.fft_ifftn, r"""
+ifftn(input, s=None, dim=None, norm=None) -> Tensor
+
+Computes the N dimensional inverse discrete Fourier transform of :attr:`input`.
+
+Args:
+    input (Tensor): the input tensor
+    s (Tuple[int], optional): Signal size in the transformed dimensions. If
+        given, the input will either be zero-padded or trimmed to this length
+        before computing the IFFT.
+    dim (Tuple[int], optional): Dimensions to be inverse Fourier transformed.
+        Default: all dimensions.
+    norm (str, optional): Normalization mode. For the backward transform
+        (:func:`~torch.fft.ifftn`), these correspond to:
+
+        * ``"forward"`` - no normalization
+        * ``"backward"`` - normalize by ``1/n``
+        * ``"ortho"`` - normalize by ``1/sqrt(n)`` (making the IFFT orthonormal)
+
+        Where ``n = prod(s)`` is the logical IFFT size.
+        Calling the forward transform (:func:`~torch.fft.fftn`) with the same
+        normalization mode will apply an overall normalization of ``1/n`` between
+        the two transforms. This is required to make :func:`~torch.fft.ifftn`
+        the exact inverse.
+
+        Default is ``"backward"`` (normalize by ``1/n``).
+
+Example:
+
+    >>> import torch.fft
+    >>> x = torch.rand(10, 10, dtype=torch.complex64)
+    >>> ifftn = torch.fft.ifftn(t)
+
+    The Fourier transform is separable, so :func:`~torch.fft.ifftn` here is
```

Comment:
Won't the normalization be different between two ffts and a single fftn, though?

Reply:
Consider an N by M transform:
- `fftn` normalizes by `N*M` directly.
- 1st `fft` normalizes by `N`, 2nd `fft` normalizes by `M`. Overall result is an `N*M` normalization.

Reply:
You can also see that NumPy implements `fftn` exactly as multiple calls to `fft` without any changes to the normalization behavior:
https://github.com/numpy/numpy/blob/3b294521a7394c2195e6b2980555c6b27258bac9/numpy/fft/_pocketfft.py#L699-L706